streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -160,26 +172,190 @@ public void shouldIdempotentlyUpdateSubscriptionFromActiveAssignment() {
     }
 
     @Test
-    public void shouldReturnOffsetsForAllCachedTaskIdsFromDirectory() throws IOException {
-        final File[] taskFolders = asList(testFolder.newFolder("0_1"),
-                                          testFolder.newFolder("0_2"),
-                                          testFolder.newFolder("0_3"),
-                                          testFolder.newFolder("1_1"),
-                                          testFolder.newFolder("dummy")).toArray(new File[0]);
+    public void tryLockForAllTaskDirectoriesShouldBeNoopIfStateDirIsEmpty() throws IOException {
+        expect(stateDirectory.listTaskDirectories()).andReturn(new File[0]).once();
+        expect(stateDirectory.lock(anyObject()))
+            .andThrow(new RuntimeException("Should not try to lock anything")).anyTimes();
 
-        assertThat((new File(taskFolders[0], StateManagerUtil.CHECKPOINT_FILE_NAME)).createNewFile(), is(true));
-        assertThat((new File(taskFolders[1], StateManagerUtil.CHECKPOINT_FILE_NAME)).createNewFile(), is(true));
-        assertThat((new File(taskFolders[3], StateManagerUtil.CHECKPOINT_FILE_NAME)).createNewFile(), is(true));
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
 
-        expect(stateDirectory.listTaskDirectories()).andReturn(taskFolders).once();
+        verify(stateDirectory);
+        assertTrue(taskManager.lockedTaskDirectories().isEmpty());
+    }
```

Comment:
I think we should actually just make `listTaskDirectories()` always return an empty `File[]` instead of null in some cases and `new File[0]` in others, as we treat both cases the same. WDYT?

Reply:
Yeah, I agree with you. I checked the code and it should be OK to always return an empty array. However, could you open a second PR for it that we merge before this one to keep this PR focussed.

Reply:
I'm not sure it's really worth doing as a separate PR, it's about 10 lines of code and is only really motivated by the work in this KIP?

Reply:
Fair enough